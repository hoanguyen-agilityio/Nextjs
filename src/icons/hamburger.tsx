import { IconProps } from '@/types';

const GoogleIcon: React.FC<IconProps> = ({ width, height, className }) => (
  <svg
    width={width}
    height={height}
    viewBox="0 0 24 24"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    className={className}
  >
    <g id="SVGRepo_bgCarrier" stroke-width="0" />

    <g
      id="SVGRepo_tracerCarrier"
      stroke-linecap="round"
      stroke-linejoin="round"
    />

    <g id="SVGRepo_iconCarrier">
      <path
        d="M4 6H20M4 12H20M4 18H20"
        stroke="#000000"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </g>
  </svg>
);

export default GoogleIcon;
