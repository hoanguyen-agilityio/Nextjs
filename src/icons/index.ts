export { default as Logo } from './Logo';
export { default as DashboardIcon } from './Dashboard';
export { default as InboxIcon } from './Inbox';
export { default as WalletIcon } from './Wallet';
export { default as NewsFeedIcon } from './NewsFeed';
export { default as AnalyticsIcon } from './Analytics';
export { default as MarketIcon } from './Market';
export { default as ArchiveIcon } from './Archive';
export { default as ProfileIcon } from './Profile';
export { default as SettingIcon } from './Settings';
export { default as PlusIcon } from './Plus';
export { default as MoonIcon } from './Moon';
export { default as NotificationIcon } from './Notification';
export { default as SearchIcon } from './Search';
export { default as UpAndDownArrows } from './UpAndDownArrows';
export { default as RightArrowUp } from './RightArrowUp';
export { default as NewCustomerIcon } from './NewCustomer';
export { default as CalendarIcon } from './Calendar';
export { default as ChartArrows } from './ChartArrows';
export { default as FilterIcon } from './Filter';
export { default as ArrowLeft } from './ArrowLeft';
export { default as ArrowRight } from './ArrowRight';
export { default as CheckedIcon } from './Checked';
export { default as ProductIcon } from './Product';
export { default as CloseIcon } from './CloseIcon';
export { default as FaceFrownIcon } from './FaceFrownIcon';
export { default as ThreeDotsIcon } from './ThreeDots';
export { default as ClockIcon } from './Clock';
export { default as RatingIcon } from './Rating';
export { default as ImageIcon } from './Image';
export { default as FileIcon } from './File';
export { default as BackIcon } from './Back';
export { default as LockIcon } from './Lock';
export { default as EyeSlashFilledIcon } from './EyeSlashFilled';
export { default as EyeFilledIcon } from './EyeFilled';
export { default as GoogleIcon } from './Google';
export { default as AppleIcon } from './Apple';
export { default as LogoWithoutText } from './LogoWithoutText';
export { default as Logout } from './Logout';
export { default as Hamburger } from './hamburger';
