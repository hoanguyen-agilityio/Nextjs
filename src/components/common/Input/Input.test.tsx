import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import InputCustom from '.';
import { SearchIcon } from '@/icons';

describe('Search component', () => {
  test('renders with default type', () => {
    const { container } = render(
      <InputCustom
        color="default"
        placeholder="search"
        size="xs"
        radius="xs"
        startContent={
          <SearchIcon
            width="21px"
            height="21px"
            className="mr-5 cursor-pointer"
          />
        }
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
