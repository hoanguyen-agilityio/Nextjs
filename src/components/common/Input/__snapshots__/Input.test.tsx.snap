// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders with default type 1`] = `
<div>
  <div
    class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end"
    data-filled="true"
    data-filled-within="true"
    data-slot="base"
  >
    <div
      class="flex flex-col w-xl h-14"
      data-slot="main-wrapper"
    >
      <div
        class="relative tap-highlight-transparent flex-row shadow-sm gap-3 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 min-h-10 transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background bg-gray-50 flex items-center space-x-2 h-full placeholder:text-ratio-300 dark:bg-gray-700 dark: text-ratio-700 dark:placeholder:text-ratio-700 focus-within:ring-0 py-4 px-6 w-full rounded-lg"
        data-slot="input-wrapper"
        style="cursor: text;"
      >
        <div
          class="w-full items-center h-full box-border flex"
          data-slot="inner-wrapper"
        >
          <svg
            class="mr-5 cursor-pointer"
            fill="none"
            height="21px"
            viewBox="0 0 22 22"
            width="21px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M10.25 20.9878C4.6 20.9878 0 16.3878 0 10.7378C0 5.08779 4.6 0.487793 10.25 0.487793C15.9 0.487793 20.5 5.08779 20.5 10.7378C20.5 16.3878 15.9 20.9878 10.25 20.9878ZM10.25 1.98779C5.42 1.98779 1.5 5.91779 1.5 10.7378C1.5 15.5578 5.42 19.4878 10.25 19.4878C15.08 19.4878 19 15.5578 19 10.7378C19 5.91779 15.08 1.98779 10.25 1.98779Z"
              fill="#809FB8"
              fill-rule="evenodd"
            />
            <path
              d="M20.5502 21.9877C20.3602 21.9877 20.1702 21.9177 20.0202 21.7677L17.0202 18.7677C16.7302 18.4777 16.7302 17.9977 17.0202 17.7077C17.3102 17.4177 17.7902 17.4177 18.0802 17.7077L21.0802 20.7077C21.3702 20.9977 21.3702 21.4777 21.0802 21.7677C20.9302 21.9177 20.7402 21.9877 20.5502 21.9877Z"
              fill="#809FB8"
            />
          </svg>
          <input
            aria-label="search"
            class="w-full bg-transparent !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text group-data-[has-value=true]:text-default-foreground placeholder:text-ratio-300 dark:placeholder:text-ratio-700 dark: text-ratio-700 flex-grow focus:outline-none focus:ring-0 text-base font-semibold"
            data-has-start-content="true"
            data-slot="input"
            id="react-aria-:r0:"
            placeholder="search"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
